### PR TITLE
Updating the portal tests

### PR DESCRIPTION
**Pull request by somewhatabstract**
After recent reworking, some of the tests were no longer testing what they used to, and they were far too complicated. This fixes all that and ups the coverage about 20% points.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 2d421c64bc562438e379571a7f7507645e8f3a5f

https://deploy-preview-179--wonder-blocks.netlify.com

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.js [hunk 1]
```diff
@@ -28,66 +28,88 @@ type Props = {|
...
 
-    _maybeDoMount() {
-        // We're not going to mount anything if we no children or we haven't
-        // rendered yet. That's because the first step in mounting the portal
-        // is to get the anchor node, which is what we render in the `render()`
-        // method.
-        const {children} = this.props;
-        if (!this._rendered || !children) {
-            return;
+    _getPortalParent() {
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.js]: Good call splitting this out.

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.js [hunk 2]
```diff
@@ -96,7 +118,7 @@ export default class TooltipPortalMounter extends React.Component<Props> {
         this._refreshPortalContent();
     }
 
-    _doUnmount() {
+    _destroyPortal() {
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.js]: Thanks for changing the names.

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js [hunk 3]
```diff
@@ -14,119 +14,160 @@ import {TooltipPortalAttributeName} from "../util/constants.js";
...
 
-import type {Placement} from "../util/types.js";
-
 type TestHarnessProps = {|
-    refChild: (Element | ?React.Component<*, *>) => mixed,
-    placement: Placement,
+    /**
+     * Identifier to give the bubble
+     */
+    bubbleId?: ?string,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js]: Do we need both `null` and `undefined`?

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js [hunk 4]
```diff
@@ -137,65 +178,75 @@ describe("TooltipPortalMounter", () => {
...
+    test("anchor changes, portal remains the same", () => {
+        // Arrange
+        const bubbleId = "the-bubble";
+        const wrapper = mount(
+            <TestHarness switchAnchor={true} bubbleId={bubbleId} />,
+        );
+        const tooltipPortal = getTooltipPortalForBubble(bubbleId);
+
+        // Act
+        wrapper.instance().forceUpdate();
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js]: I was wondering how the anchor switching... I've found my answer.  Cool!

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js [hunk 5]
```diff
@@ -137,65 +178,75 @@ describe("TooltipPortalMounter", () => {
...
-
-        arrangeAct(andAssert);
+        const bubbleId = "the-bubble";
+        const wrapper = mount(<TestHarness bubbleId={bubbleId} />);
+        const tooltipPortal = getTooltipPortalForBubble(bubbleId);
+        expect(tooltipPortal).toBeTruthy();
+
+        // Act
+        wrapper.setProps({hasAnchor: false});
+        // wrapper.instance().forceUpdate();
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js]: delete?

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js [hunk 6]
```diff
@@ -14,119 +14,160 @@ import {TooltipPortalAttributeName} from "../util/constants.js";
...
-    updateAnchorRef(node) {
-        if (node) {
-            const element = ReactDOM.findDOMNode(node);
-            if (this.state.anchor !== element) {
-                this.setState({anchor: ((element: any): ?HTMLElement)});
-            }
+        if (this._lastAnchor === "anchorA") {
+            return <View key={this._lastAnchor}>Anchor A</View>;
+        } else {
+            return <HeadingSmall key={this._lastAnchor}>Anchor B</HeadingSmall>;
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js]: Interesting way of forcing prop updates.

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js (hunk 3 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js]: Removed this flexibility. No longer needed.

File: packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js (hunk 6 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js]: Yeah, may not be the best, but it was the first thing that came to mind.

**Pull request merged**